src/mumble/JackAudio.cpp

```diff
@@ -1059,6 +1094,77 @@ bool JackAudioOutput::process(const jack_nframes_t frames) {
 	return true;
 }
 
+void JackAudioOutput::prepareOutputBuffers(unsigned int frameCount, QList< AudioOutputBuffer * > *qlMix,
+										   QList< AudioOutputBuffer * > *qlDel) {
+	ServerHandlerPtr sh = Global::get().sh;
+	VoiceRecorderPtr recorder;
+	if (sh) {
+		recorder = Global::get().sh->recorder;
+	}
+
+	// Keep track of ports we've filled the audio of.
+	QList< jack_port_t * > qlPortsYetToFill(userPorts.values());
+
+	// Register user ports based on who is currently present in qmOutputs and route their audio to JACK.
+	// Don't care if users get removed for now. TODO: maybe at some point we do.
+	QMultiHash< const ClientUser *, AudioOutputBuffer * >::const_iterator it = qmOutputs.constBegin();
+	while (it != qmOutputs.constEnd()) {
+		const ClientUser *user   = it.key();
+		AudioOutputBuffer *audio = it.value();
+
+		if (!user || !recorder || (recorder && !recorder->isTransportEnabled())) {
+			if (audio->prepareSampleBuffer(frameCount)) {
+				qlMix->append(audio);
+			} else {
+				qlDel->append(audio);
+			}
+
+			++it;
+			continue;
+		}
+
+		QString qsPortName = user->qsName;
+		qsPortName         = qsPortName.prepend("user_");
+		if (!userPorts[qsPortName]) {
+			auto port = jas->registerPort(qsPortName.toStdString().c_str(), JackPortIsOutput);
```

Comment:
Use of `auto` does not conform to coding guidelines

Reply:
Will fix.